test/NuGet.Clients.Tests/NuGet.CommandLine.Test/FileSystemBackedV3MockServer.cs

```diff
@@ -128,6 +128,30 @@ private Action<HttpListenerResponse> ServerHandlerV3(HttpListenerRequest request
                         });
                     }
                 }
+                else if (path.StartsWith("/packages/"))
```

Comment:
Is this `/packages` path where newly created packages go by default or why it's so specific?

Reply:
The following are what I learnt, I hope it would help.
`FileSystemBackedV3MockServer` is a mock Server used for testing. When we create a mock server and add the service index Uri into our testing NuGet.Config file, then running NuGet.exe install, the mock server will respond like a real V3 http package source.
However, the `ServerHandlerV3` method in `FileSystemBackedV3MockServer` is not complete to run my newly added tests. 
When NuGet.exe tries the registration URL first, it will [BuildRegistrationIndexResponse](https://github.com/NuGet/NuGet.Client/blob/dev-hengliu-CM-unlisted/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/FileSystemBackedV3MockServer.cs#L119), and [CreatePackageRegistrationBlob](https://github.com/NuGet/NuGet.Client/blob/dev-hengliu-CM-unlisted/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MockResponses/MockResponseBuilder.cs#L189), and it will set `packageContent ` blob as ["packageContent", $"{server.Uri}packages/{id}.{version}.nupkg"](https://github.com/NuGet/NuGet.Client/blob/dev-hengliu-CM-unlisted/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/Util.cs#L765) 

During NuGet.exe install, package needs to be downloaded, the `DownloadResourceV3.cs` will finally get the [packageContent Uri ](https://github.com/NuGet/NuGet.Client/blob/dev-hengliu-CM-unlisted/src/NuGet.Core/NuGet.Protocol/Resources/DownloadResourceV3.cs#L118), which is `$"{server.Uri}packages/{id}.{version}.nupkg`. So that why I need to add the logic to handle `path.StartsWith("/packages/")`. Only in this way, the test of install could be completed, or else, the download package could not be done.

Reply:
I think this server should mimic what a proper v3 server would do. 
Can we change https://github.com/NuGet/NuGet.Client/blob/7e23737fdf89fe34c90ac8878c96843b2f0dd17e/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/Util.cs#L765 instead?

Reply:
In particular look what https://api.nuget.org/v3/registration5-semver1/newtonsoft.json/index.json `packageContent` points to.

Reply:
Thanks, Nikolche!
I see `"packageContent":"https://api.nuget.org/v3-flatcontainer/newtonsoft.json/3.5.8/newtonsoft.json.3.5.8.nupkg"`
So change the line to `item.Add(new JProperty("packageContent", $"{server.Uri}v3-flatcontainer/{id}/{version}/{id}.{version}.nupkg")); `, right?
But we still need to add logic to handle path.StartsWith("/v3-flatcontainer/") and seems there will be lots of tests needs to be fixed. 
I also found 100+ packageContent with "`/packages/`" in [JsonData.cs](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/JsonData.cs#L1047
) Those need to be changed as well if changing this line, right?https://github.com/NuGet/NuGet.Client/blob/7e23737fdf89fe34c90ac8878c96843b2f0dd17e/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/Util.cs#L765

Reply:
Flat is handled in https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/FileSystemBackedV3MockServer.cs#L58. 

You can chang that to v3-flatcontainer if you want, and then change the index.json that it advertises.

Reply:
Thanks! Only changed the package content format in util works. 
No other change is needed. So my understanding of changing 4 parts is incorrect.